src/app/core/registry/registry.service.ts

```diff
@@ -495,4 +556,19 @@ export class RegistryService {
       }
     });
   }
+
+  queryMetadataFields(query: string): Observable<RemoteData<PaginatedList<MetadataField>>> {
+    /**
+     * This should come directly from the server in the future
```

Comment:
I've talked this over with @benbosman, and we concluded that this type of search is much easier to do in the UI than it would be in the backend. So we decided to keep it here. We'll remove this comment.

Reply:
Sounds good, thanks!